sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -298,6 +299,21 @@ object ResolveDefaultColumns extends QueryErrorsBase
     val analyzed: Expression = plan.collectFirst {
       case Project(Seq(a: Alias), OneRowRelation()) => a.child
     }.get
+
+    if (!analyzed.foldable) {
+      throw QueryCompilationErrors.defaultValueNotConstantError(statementType, colName, defaultSQL)
+    }
+
+    // Another extra check, expressions should already be resolved if AnalysisException is not
+    // thrown in the code block above
+    if (!analyzed.resolved) {
+      throw QueryCompilationErrors.defaultValuesUnresolvedExprError(
```

Comment:
- https://github.com/apache/spark/actions/runs/9535478067/job/26281266890:
```
[error] /home/runner/work/spark/spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala:299:36: value defaultValueNotConstantError is not a member of object org.apache.spark.sql.errors.QueryCompilationErrors
[error]       throw QueryCompilationErrors.defaultValueNotConstantError(statementType, colName, defaultSQL)
[error]                         
```
@urosstan-db @cloud-fan Seems that branch-3.5 does not have the `QueryCompilationErrors.defaultValuesUnresolvedExprError` method, which caused the branch-3.5 daily build to fail.

Reply:
should we backport https://github.com/apache/spark/pull/41345 or change to use an existing function?

Reply:
https://github.com/apache/spark/pull/46978 is the backport

Reply:
thanks @cloud-fan